cmake/configure/configure_boost.cmake

```diff
@@ -136,6 +136,54 @@ MACRO(FEATURE_BOOST_FIND_EXTERNAL var)
       RESET_CMAKE_REQUIRED()
     ENDIF()
 
+    IF(${var})
+      # We want to pass the libraries to the CMake project that tests for the
+      # BOOST Serialization bug. If we don't change the separator, only the
+      # first one is passed. This change in the separator is reverted in
+      # TestBoostBug/CMakeLists.txt
+      STRING (REPLACE ";" "|" BOOST_LIBRARIES_SEPARATED "${BOOST_LIBRARIES}")
```

Comment:
btw, why do you need to do all those tricks? Can't you use the found boost directly and add targets right here and build them to see whether the test pass? I mean essentially use the result of CMake's find directly.

The might be some hidden reasons why passing the include dirs and libs as you do now does not work on macOS.

Reply:
I am happy to take any improvement. What exactly would you do instead?

In fact, I can reproduce these failures now using [Travis](https://travis-ci.org/masterleinad/TestBoostBug/jobs/313069473) and a regular setup (using FIND_PACKAGE(Boost)) in https://github.com/masterleinad/TestBoostBug

Reply:
> I am happy to take any improvement. What exactly would you do instead?

that's the problem -- i don't know :-)

As for `rpaths`, all i have in my own cmake is
```
#
# RPATHS: https://cmake.org/Wiki/CMake_RPATH_handling#Mac_OS_X_and_the_RPATH
#
IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
  ## cmake  --help-policy CMP0042
  set(CMAKE_MACOSX_RPATH ON)
ENDIF()

# use, i.e. don't skip the full RPATH for the build tree
SET(CMAKE_SKIP_BUILD_RPATH  FALSE)

# when building, don't use the install RPATH already
# (but later on when installing)
SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)

SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")

# add the automatically determined parts of the RPATH
# which point to directories outside the build tree to the install RPATH
SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)

# the RPATH to be used when installing, but only if it's not a system directory
LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
IF("${isSystemDir}" STREQUAL "-1")
   SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
ENDIF("${isSystemDir}" STREQUAL "-1")

# make sure RPATHs are used on macOS
SET(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
```
But frankly, i am also not sure if any/all of that is required...

Reply:
p.s. you repo seems to work fine for me with `cmake ../ -DCMAKE_PREFIX_PATH=<boost_prefix>`.

Reply:
Maybe that means that my latest change finally worked. Let's see what Travis says.